src/js/actions/tools.js

```diff
@@ -985,21 +1002,28 @@ define(function (require, exports) {
             key: utilShortcuts.GLOBAL.TOOLS.MASK_SELECT,
             modifiers: {},
             fn: _vectorMaskHandler,
-            name: "Vector Mask"
+            name: nls.localize("strings.TOOLS.NAMES.VECTOR_MASK_MODE")
         });
 
         _vectorSelectMaskHandler = function () {
             if (toolStore.getVectorMode()) {
-                flux.actions.tools.select(toolStore.getToolByID("superselectVector"));
+                flux.actions.tools.select(toolStore.getToolByID(directSelectToolID));
             }
         }.bind(this);
 
-        shortcutSpecs.push({
-            key: utilShortcuts.GLOBAL.TOOLS.VECTOR_SELECT,
-            modifiers: {},
-            fn: _vectorSelectMaskHandler,
-            name: "Super Select Vector"
-        });
+        if (toolStore.getVectorMode()) {
+            _vectorSelectMaskHandler = function () {
+                flux.actions.select(toolStore.getToolByID(directSelectToolID));
+            }.bind(this);
+
+            shortcutSpecs.push({
+                key: utilShortcuts.GLOBAL.TOOLS.VECTOR_SELECT,
+                modifiers: {},
+                id: directSelectToolID,
+                fn: _vectorSelectMaskHandler,
+                name: nls.localize("strings.TOOLS.NAMES.DIRECT_SELECTION_TOOL")
```

Comment:
It depends on the vector mask state and all of that is being calculated in `tools.js`, so that is why it is being added separately after we check for `toolStore.getVectorMode()`.